lib/sycamore/sycamore/utils/time_trace.py

```diff
@@ -4,8 +4,59 @@
 import resource
 import threading
 import functools
+import logging
 from sys import platform
 
+zero_time = time.time_ns()
+# Mac reports ru_maxrss in bytes. Linux reports it in kB.
+if platform == "darwin":
+    RSS_MULTIPLIER = 1
+else:
+    RSS_MULTIPLIER = 1024
+
+
+class TimeTraceData:
+    def __init__(self, t0, t1, user, sys, rss):
+        self.t0 = t0  # ns
+        self.t1 = t1  # ns
+        self.user = user  # ns
+        self.sys = sys  # ns
+        self.rss = rss  # bytes
+
+    def wall_s(self):
+        return (self.t1 - self.t0) / 1.0e9
```

Comment:
It seems you're not a fan of `1000000000.0`.  Not everybody likes scientific notation, either.  Perhaps it's time for some constant like `BILLION` or `NANOS_IN_ONE`.

Reply:
Yes, because it's hard to tell if its 100000000.0. Do you object to scientific?  It's compact and explicit. I can change it to a constant although I don't understand why that's better.